time_weighted_average_price/on_chain/README.md

```diff
@@ -1,32 +1,107 @@
-# On Chain
+# Bringing An Attestation Of Time-Weighed Average Price On Chain
 
-This demo shows how we can get transitive attested data on chain. In this demo,
-bring a transitive attested function call from the [twap demo](../attest_fn_call)
-on to a chain in a development environment.
+This example shows you how to bring an attestation of time-weighted average price (TWAP)
+on chain. It builds on the 
+[Attesting A Time-Weighted Average Price](../attest_fn_call) 
+example, which shows you how to obtain an attestation of a TWAP, and on the
+[Hello World - Bringing A Blocky AS Function Call Attestation On Chain](../../hello_world_on_chain)
+example, which shows you how to bring an attestation of a Blocky AS function
+call on chain.
+
+In this example, you'll learn how to:
+
+- Bring an attestation of a TWAP on chain 
+- Revert a smart contract transaction if the 
 
 ## Setup
 
 Set up the project dependencies:
 
 ```bash
-npm install
+npm install --dd
 ```
 
-## Run
+## Quick Start
 
-Test bringing a transitive attested function call on chain into the
-[User](contracts/User.sol) contract:
+To run this example, call:
 
 ```bash
 make test-local
 ```
 
-You can see the details of this test in [test/user.ts](test/user.ts).
+which will test verifying an attested TWAP in
+[`contracts/User.sol`](contracts/User.sol)
+within a local test environment and give your output like:
 
-(Optionally) If you ran the [twap demo](../attest_fn_call) and want to use its
-latest transitive attested function call, bring it into this project by
-running:
+```
+  Local Tests
+    ✔ Verify attested TWAP in User contract (1080ms)
+```
 
-```bash
-make copy-from-twap
+## Walkthrough
```

Comment:
I think we can lead on the previous example in this walkthough a bit, but let me know if you think I'm going too far